src/cli/cmd/add/flake.rs

```diff
@@ -148,23 +148,35 @@ impl AttrType {
         flake_input_name: &str,
         flake_input_value: &url::Url,
     ) -> color_eyre::Result<String> {
-        let mut added_cosmetic_newline = false;
+        match self {
+            AttrType::Inputs(ref inputs_attr) => {
+                match inputs_attr.from.len() {
+                    // inputs = { nixpkgs.url = ""; };
+                    1 => {
+                        let first_input = find_attrset(&inputs_attr.to, None)?.expect("");
+                        let (from_span, _to_span) = Self::kv_to_span(&first_input);
+                        let flake_input =
+                            format!(r#"{flake_input_name}.url = "{flake_input_value}";{NEWLINE}"#);
+                        AttrType::insert_input(from_span, flake_contents, &flake_input, false)
+                    }
 
-        // We don't do anything fancy like trying to match the existing format of e.g.
-        // `inputs = { <input_name>.url = "<input_value>"; };`
-        let mut flake_input =
-            format!(r#"inputs.{flake_input_name}.url = "{flake_input_value}";{NEWLINE}"#);
+                    // inputs.nixpkgs = { url = ""; inputs.something.follows = ""; };
+                    // OR
+                    // inputs.nixpkgs.url = "";
+                    2 | 3 => {
+                        let (from_span, _to_span) = self.span();
 
-        match self {
-            AttrType::Inputs(_) => {
-                let (from_span, _to_span) = self.span();
+                        let flake_input = format!(
+                            r#"inputs.{flake_input_name}.url = "{flake_input_value}";{NEWLINE}"#
+                        );
+                        AttrType::insert_input(from_span, flake_contents, &flake_input, false)
+                    }
 
-                AttrType::insert_input_above_span(
-                    from_span,
-                    flake_contents,
-                    &flake_input,
-                    added_cosmetic_newline,
-                )
+                    // I don't think this is possible, but just in case.
+                    len => Err(color_eyre::eyre::eyre!(
```

Comment:
May be wise to emit a warning asking for a bug report, then default to creating the input using our "default" style to get it working for the user?

Reply:
Nice idea.

Reply:
Great idea, even; I'll make this part of the "setup tracing" ticket though :+1: